Mathlib/Analysis/SpecialFunctions/Log/PosLog.lean

```diff
@@ -0,0 +1,160 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+import Mathlib.Analysis.SpecialFunctions.Log.Basic
+
+/-!
+# The Positive Part of the Logarithm
+
+This file defines the function `log⁺ = r ↦ max 0 (log r)`, and establishes
+standard estimates.
+-/
+
+namespace Real
+
+/-!
+## Definition, Notation and Reformulations
+-/
+/-- Definition: the positive part of the logarithm. -/
+noncomputable def posLog : ℝ → ℝ := fun r ↦ max 0 (log r)
+
+/-- Notation `log⁺` for the real part of the logarithm. -/
+scoped notation "log⁺" => posLog
+
+/-- Definition of the positive part of the logarithm, formulated as a theorem. -/
+theorem posLog_def {r : ℝ} : log⁺ r = max 0 (log r) := rfl
+
+/-!
+## Elementary Properties
+-/
+/-- Presentation of `log` in terms of its positive part. -/
+theorem posLog_sub_posLog_inv {r : ℝ} : log⁺ r - log⁺ r⁻¹ = log r := by
+  rw [posLog_def, posLog_def, log_inv]
+  by_cases h : 0 ≤ log r
+  · simp [h]
+  · rw [not_le] at h
+    simp [neg_nonneg.1 (Left.nonneg_neg_iff.2 h.le)]
+
+/-- Presentation of `log⁺` in terms of `log`. -/
+theorem half_mul_log_add_log_abs {r : ℝ} : 2⁻¹ * (log r + |log r|) = log⁺ r := by
+  by_cases hr : 0 ≤ log r
+  · simp [posLog, hr, abs_of_nonneg]
+    ring
+  · simp [posLog, le_of_not_ge hr, abs_of_nonpos]
+
+/-- The positive part of `log` is never negative. -/
+theorem posLog_nonneg {x : ℝ} : 0 ≤ log⁺ x := by simp [posLog]
+
+/-- The function `log⁺` is even. -/
+@[simp]
+theorem posLog_neg (x : ℝ) : log⁺ (-x) = log⁺ x := by simp [posLog]
+
+/-- The function `log⁺` is even. -/
+theorem posLog_abs (x : ℝ) : log⁺ |x| = log⁺ x := by simp [posLog]
+
+/-- The function `log⁺` is zero in the interval [-1,1]. -/
+theorem posLog_eq_zero_iff (x : ℝ) : log⁺ x = 0 ↔ |x| ≤ 1 := by
+  rw [← posLog_abs, ← log_nonpos_iff (abs_nonneg x)]
+  simp [posLog]
+
+/-- The function `log⁺` equals `log` outside of in the interval (-1,1). -/
+theorem posLog_eq_log {x : ℝ} (hx : 1 ≤ |x|) : log⁺ x = log x := by
+  simp [posLog]
+  rw [← log_abs]
+  apply log_nonneg hx
+
+/-- The function `log⁺` equals `log` for all natural numbers. -/
+theorem log_of_nat_eq_posLog {n : ℕ} : log⁺ n = log n := by
+  by_cases hn : n = 0
+  · simp [hn, posLog]
+  · simp [posLog_eq_log, Nat.one_le_iff_ne_zero.2 hn]
+
+/-- The function `log⁺` is monotone in the positive axis. -/
+theorem monotoneOn_posLog : MonotoneOn log⁺ (Set.Ici 0) := by
+  intro x hx y hy hxy
+  simp [posLog]
```

Comment:
nonterminal simp. Replace with the output of `simp? [posLog]`.

Reply:
Done.